core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1231,10 +1238,10 @@ class UnifiedLog(@volatile var logStartOffset: Long,
           }
 
           remoteLogManager.get.findOffsetByTimestamp(topicPartition, targetTimestamp, logStartOffset, leaderEpochCache.get)
-        } else None
+        } else Optional.empty()
```

Comment:
We should return None, right?

Reply:
Earlier `remoteLogManager#findOffsetByTimestamp` was returning Scala `Option` but it returns `Optional` now after it is migrated to Java.